src/listenMqtt.js

```diff
@@ -0,0 +1,443 @@
+		case "AdminTextMessage":
+			switch (v.delta.type) {
+				case "change_thread_theme":
+				case "change_thread_nickname":
+				case "change_thread_icon":
+					break;
+				case "group_poll":
+					var fmtMsg;
+					try {
+						fmtMsg = utils.formatDeltaEvent(v.delta);
+					} catch (err) {
+						return globalCallback({
+							error: "Problem parsing message object. Please open an issue at https://github.com/Schmavery/facebook-chat-api/issues.",
+							detail: err,
+							res: v.delta,
+							type: "parse_error"
+						});
+					}
+					return globalCallback(null, fmtMsg);
+					break;
+				default:
+					return;
+			}
+		//For group images
+		case "ForcedFetch":
+			var mid = v.delta.messageId;
+			var tid = v.delta.threadKey.threadFbId.toString();
+			const form = {
+				"av": ctx.globalOptions.pageID,
+				"queries": JSON.stringify({
+					"o0": {
+						"doc_id": "1768656253222505",
+						"query_params": {
+							"thread_and_message_id": {
+								"thread_id": tid,
+								"message_id": mid,
+							}
+						}
+					}
+				})
+			};
+
+			defaultFuncs
+				.post("https://www.facebook.com/api/graphqlbatch/", ctx.jar, form)
+				.then(utils.parseAndCheckLogin(ctx, defaultFuncs))
+				.then((resData) => {
+					if (resData[resData.length - 1].error_results > 0) {
+						throw resData[0].o0.errors;
+					}
+
+					if (resData[resData.length - 1].successful_results === 0) {
+						throw { error: "forcedFetch: there was no successful_results", res: resData };
+					}
+
+					var fetchData = resData[0].o0.data.message;
+
+					globalCallback(null, {
+						type: "change_thread_image",
+						threadID: utils.formatID(tid.toString()),
+						snippet: fetchData.snippet,
+						timestamp: fetchData.timestamp_precise,
+						author: fetchData.message_sender.id,
+						image: {
+							attachmentID: fetchData.image_with_metadata.legacy_attachment_id,
+							width: fetchData.image_with_metadata.original_dimensions.x,
+							height: fetchData.image_with_metadata.original_dimensions.y,
+							url: fetchData.image_with_metadata.preview.uri
+						  }
+					  });
+				})
+				.catch((err) => {
+					log.error("forcedFetch", err);
+				});
+			break;
+		case "ThreadName":
+		case "ParticipantsAddedToGroupThread":
+		case "ParticipantLeftGroupThread":
+			var formattedEvent;
+			try {
+				formattedEvent = utils.formatDeltaEvent(v.delta);
+			} catch (err) {
+				return globalCallback({
+					error: "Problem parsing message object. Please open an issue at https://github.com/Schmavery/facebook-chat-api/issues.",
+					detail: err,
+					res: v.delta,
+					type: "parse_error"
+				});
+			}
+			return (!ctx.globalOptions.selfListen &&
+				formattedEvent.author.toString() === ctx.userID) ||
+				!ctx.loggedIn ?
+				undefined :
+				globalCallback(null, formattedEvent);
+	}
+}
+
+module.exports = function (df, api, ctx) {
+	var currentlyRunning = null;
+	var globalCallback = identity;
+	defaultFuncs = df;
+	return function (callback) {
```

Comment:
I find this function a bit confusing, but tbf it's been a while since I spent time with the old listen code. Is this the stopListening callback? it seems to do a lot of stuff, whereas the listenMqtt function itself doesn't seem to do anything.
Maybe I'm reading this wrong.

Reply:
This is the function itself that you pass another callback function to it to recieve messages and events.
stopListening is not there yet but I'll try to add it too.

Reply:
This is the function itself that you pass another callback function to it to recieve messages and events.
stopListening is not there yet but I'll try to add it too.

Reply:
Added stopListening.